### DISCUSSION TITLE
Can't get nested array but returns an object instead

### DISCUSSION THREAD

**Discussion by kylemiller3** [prisma/prisma#3254] [Q&A] [answered]
I have this schema:

```prisma
model organizations {
  id                        Int                         @default(autoincrement()) @id @map("_id")
...
  user_organization_xref    user_organization_xref[]
}

model user_organization_xref {
...
  organizations       organizations @relation(fields: [fk_organization_id], references: [id])
  users               users         @relation(fields: [fk_user_id], references: [id])

  @@id([fk_user_id, fk_organization_id])
}

model users {
  id                        Int                      @default(autoincrement()) @id @map("_id")
...
  user_organization_xref    user_organization_xref[]
}
```
And my resolver:
```typescript
    user: async (_parent, _args: {email: string}, { auth, prisma }: {auth: any, prisma: PrismaClient}) => {
      const user = await prisma.users.findOne({
        where: {
          email: _args.email
        }
      });
      const organizations = await prisma.user_organization_xref.findMany({
        where: {
          fk_user_id: user?.id
        },
        include: {
          users: true,
        },
      });
      const obj = { ...user, organizations };
      return obj;
    },
```

I am getting an object that looks like this:
```json
{
  "id": 1,
  "organizations": [
    {
      "fk_user_id": 1,
      "fk_organization_id": 1,
      "users": {
        "id": 1,
      }
    }
  ]
}
```

But I am expecting an object that looks like this: 
```json
{
  "id": 1,
  "organizations": [
    {
      "fk_user_id": 1,
      "fk_organization_id": 1,
      "users": [
        {
          "id": 1,
        }
      ]
    }
  ]
}
```

Why do I get an object instead of an array and how do I fix it?

**Comment by ryands17** [comment, marked as answer]
Hey @kylemiller3 👋 

```prisma
model user_organization_xref {
...
  organizations       organizations @relation(fields: [fk_organization_id], references: [id])
  users               users         @relation(fields: [fk_user_id], references: [id])

  @@id([fk_user_id, fk_organization_id])
}
```

It's because you have specified yourself that this model only has a user, so it would return an object, not an array, which is why only a single is allowed.

If you need to fetch all the users in an organization, you would need to do this:

```ts
 await prisma.organizations.findOne({
    where: {
      id: 1,
    },
    select: {
      user_organization_xref: {
        select: {
          users: true,
        },
      },
    },
  })
```

This is a many-to-many relationship, so directly fetching the users wouldn't be possible, you would need to access them via the join table which is what I wrote above.
Let me know if this works.

**Reply by kylemiller3** [reply]
Thank you that is close to what I wanted.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
